apps/docs/pages/guides/platform/custom-domains.mdx

```diff
@@ -18,139 +18,213 @@ Custom domains allow you to present a branded experience to your users. Custom d
+- TXT record for `_cf-custom-hostname.<domain>`.
+- TXT record for `<domain>`.
+
+Use the [`domains reverify`](/docs/reference/cli/supabase-domains-reverify) command to begin the verification process of your domain. You may need to run this command a few times as DNS records take a while to propagate.
 
 ```bash
-supabase domains reverify --project-ref foobarbaz --experimental
+supabase domains reverify --project-ref abcdefghijklmnopqrst
 ```
 
-You might need to wait a few minutes before your updated DNS records are propagated, especially if the older records were using a high TTL.
+In the background, Supabase will check your DNS records and use [Let's Encrypt](https://letsencrypt.org) to issue a SSL certificate for your domain. This process can take a while, sometimes up to 30 minutes.
 
-### Activate your domain
+### Prepare to activate your domain
 
-The final activation step reconfigures your project to start serving traffic on your custom domain (`api.example.com`).
-The auth service, in particular, will no longer work with the original URL (`foobarbaz.supabase.co`).
-As such, it is recommended that you schedule a downtime window of 20-30 minutes, depending on the complexity of your project, to update all the services that need to know about your custom domain:
+Before you activate your domain, you need to understand the consequences and behavior of your Supabase project.
 
-- Any client code (e.g., frontends, mobile apps). You can initialize a new Supabase client with your custom domain, `api.example.com`. For example:
+- The project's Supabase domain remains active and will not go away.
+  - You do not need to change the Supabase URL in your applications immediately or at once.
+  - You can use it interchangeably with the custom domain.
+- Supabase Auth will use the custom domain immediately once activated.
+  - OAuth flows will advertize the custom domain as a callback URL.
+  - SAML will use the custom domain instead. This means that the `EntityID` of your project has changed, and this may cause SAML with existing identity providers to stop working.
 
-  ```js
-  import { createClient } from '@supabase/supabase-js'
+To prevent issues for your users, make sure you have gone through these steps:
 
-  // Use a custom domain as the supabase URL
-  const supabase = createClient('https://api.example.com', 'public-anon-key')
-  ```
+1. Go through all of your Supabase OAuth providers:
+   - In the provider's developer console (not in the Supabase dashboard!), find the OAuth application and add the custom domain Supabase Auth callback URL **in addition to the Supabase project URL.** Example:
+     - `https://abcdefghijklmnopqrst.supabase.co/auth/v1/callback` **and**
+     - `https://api.example.com/auth/v1/callback`
+   - [Sign in with Twitter](/docs/guides/auth/social-login/auth-twitter) uses cookies bound to the project's domain. In this case make sure your frontend code uses the custom domain instead of the default project's domain.
```

Comment:
```suggestion
   - [Sign in with Twitter](/docs/guides/auth/social-login/auth-twitter) uses cookies bound to the project's domain. Make sure your frontend code uses the custom domain instead of the default project's domain.
```